include/LCHelpers/ClusterHelper.h

```diff
@@ -94,14 +94,14 @@ class ClusterHelper
      *  @brief  Get the distance of closest approach between the projected track direction at calorimeter and the hits within a cluster.
      *          Note that only a specified number of layers are examined.
      * 
-     *  @param  pTrack address of the track
+     *  @param  trackState trackState to use for track-cluster distance calculation
      *  @param  pCluster address of the cluster
      *  @param  maxSearchLayer the maximum pseudolayer to examine
      *  @param  parallelDistanceCut maximum allowed projection of track-cluster separation along track direction
      *  @param  minTrackClusterCosAngle min cos(angle) between track and cluster initial direction
      *  @param  trackClusterDistance to receive the track cluster distance
      */
-    static pandora::StatusCode GetTrackClusterDistance(const pandora::Track *const pTrack, const pandora::Cluster *const pCluster,
+    static pandora::StatusCode GetTrackClusterDistance(const pandora::TrackState &trackState, const pandora::Cluster *const pCluster,
         const unsigned int maxSearchLayer, const float parallelDistanceCut, const float minTrackClusterCosAngle, float &trackClusterDistance);
 
```

Comment:
Might want to add three specific functions, with first argument being a Track (assuming we're going to use the TrackState at the calorimeter), a TrackState or a vector of TrackStates, with only the single TrackState having the real implementation (other options redirecting to this).

If you prefer, you could template this, with all input arguments being given by address and a single template prototype GetTrackClusterDistance(const T *const pT, ...) and three specialisations providing implementation?

Reply:
I have implemented this.